tests/test_qasm.py

```diff
@@ -134,3 +134,16 @@ def test_export_import():
 
     for u0, u1 in zip(props, read_props):
         assert (u0 - u1).norm() < 1e-12
+
+
+def test_read_qasm():
+    filename = "w-state.qasm"
+    filepath = Path(__file__).parent / 'qasm_files' / filename
+    filename2 = "w-state_with_comments.qasm"
+    filepath2 = Path(__file__).parent / 'qasm_files' / filename2
+    try:
+        qc = read_qasm(filepath)
+        qc2 = read_qasm(filepath2)
+        assert True
+    except SyntaxError:  # inline comments caused SyntaxError
+        assert False
```

Comment:
There is no need to use try except here. If an error is raised during the execution, pytest will catch it and report the error with the corresponding error message. If the error is caught manually and replaced by assert (which throw an error if False), it actually hides the true error message.